src/services/fileStorage/proxy-service.js

```diff
@@ -557,21 +572,28 @@ const renameService = {
 
 		return canWrite(userId, _id)
 			.then(() => FileModel.findOne({ _id }).exec())
-			.then((directory) => {
-				if (!directory) return Promise.reject(new NotFound('The given directory/file was not found!'));
+			.then((obj) => {
+				if (!obj) return Promise.reject(new NotFound('The given directory/file was not found!'));
 				return FileModel.update({ _id }, { name: newName }).exec();
+			})
+			.catch((e) => {
+				logger.error(e);
+				return new Forbidden();
 			});
 	},
 };
 
 const fileTotalSizeService = {
 
+	docs: swaggerDocs.fileTotalSizeService,
+
 	/**
      * @returns total file size and amount of files
-     * @param payload contains fileStorageType and userId and schoolId, set by middleware
+	 * FIX-ME:
+	 * - Check if user in payload is administrator
      */
-	find({ payload }) {
-		return FileModel.find({ owner: payload.schoolId }).exec()
+	find() {
+		return FileModel.find({}).exec()
```

Comment:
Please use `lean()` if the Mongoose extensions to the document are not needed.

Also: these are all files of the instance, or am I mising something?

Reply:
Yes all files. This is purely for the administrator view to show how much diskspace is occupied by the files of the school.

Reply:
And of course `lean()` should be used.